### PR TITLE
[Documentation] Added package-info documentation for (most) packages

### PR DESCRIPTION
**Pull request by YeffyCodeGit**
# Added package-info files for (most) classes
Related to #61

## Additions
- Added package-info files for most, if not all packages

## Notes
- I didn't write all the package-info files, because I couldn't properly understand some of the packages. If you could provide some insight on what those packages do, and /or some ideas for what to write, that would be great! :D

### CODE REVIEW

**Review by lucasstarsz — CHANGES_REQUESTED** [review]
Thanks for contributing to FastJ -- we're happy to have you!
Unfortunately, these will not work as package comments for a few reasons outlined below:

1. The style of writing overall needs improvement -- rather than try to over-explain what a package is and does, you can simply write what the package is about. Unless more clarification is needed, this is all that would be required of a `package-info` file. An example of this is provided in the comment made on `tech.fastj.engine.internals`.

Furthermore, none of these are documentation comments. A documentation comment consists of **2** asterisks (\*) after the forward slash, like so:

```java
/**
 * content
 */
```
or:
```java
/** content */
```

As for not knowing what some of these packages do, that's perfectly fine! Just focus on the packages which you actually understand the purpose of -- other people can take care of the rest.

File: src/main/java/tech/fastj/engine/internals/package-info.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package tech.fastj.engine.internals;
+
+/*
+ * This package holds classes which are meant to support FastJ from the backend only. The classes in this package are not meant
+ * to be used by the users of the library, more so the developers.
+ */
```
[the remark this answers is not in the document]
Comment (YeffyCodeGit) [reply src/main/java/tech/fastj/engine/internals/package-info.java:6]: Hey! Thanks for the pointers, i'll update it and commit again by today!

**Comment by YeffyCodeGit** [comment]
Fixed!

File: src/main/java/tech/fastj/systems/fio/package-info.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package tech.fastj.systems.fio;
+
+/**
+ * Code which provides simple methods to work with files.
```
Comment (lucasstarsz) [inline src/main/java/tech/fastj/systems/fio/package-info.java:4]: Replace "code which provides" with "system providing".

File: src/main/java/tech/fastj/systems/behaviors/package-info.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package tech.fastj.systems.behaviors;
+
+/**
+ * Code for the FastJ behaviour system.
```
Comment (lucasstarsz) [inline src/main/java/tech/fastj/systems/behaviors/package-info.java:4]: Consider: "API for FastJ's behavior system."

> behaviour

Furthermore, please use American-style wording for any documentation/code in FastJ.

File: src/main/java/tech/fastj/systems/audio/package-info.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package tech.fastj.systems.audio;
+
+/**
+ * The FastJ audio system.
```
Comment (lucasstarsz) [inline src/main/java/tech/fastj/systems/audio/package-info.java:4]: Consider: "FastJ's audio system".

File: src/main/java/tech/fastj/math/package-info.java [hunk 5]
```diff
@@ -0,0 +1,5 @@
+package tech.fastj.math;
+
+/**
+ * Math helpers meant to be used in FastJ Games.
```
Comment (lucasstarsz) [inline src/main/java/tech/fastj/math/package-info.java:4]: Consider: "FastJ's math library".

File: src/main/java/tech/fastj/input/package-info.java [hunk 6]
```diff
@@ -0,0 +1,5 @@
+package tech.fastj.input;
+
+/**
+ * Handles all input events in FastJ.
```
Comment (lucasstarsz) [inline src/main/java/tech/fastj/input/package-info.java:4]: To keep consistent with (most of) your other package info messages, remove "in FastJ".

File: src/main/java/tech/fastj/graphics/ui/package-info.java [hunk 7]
```diff
@@ -0,0 +1,5 @@
+package tech.fastj.graphics.ui;
+
+/**
+ * All the code for UI in FastJ.
```
Comment (lucasstarsz) [inline src/main/java/tech/fastj/graphics/ui/package-info.java:4]: Consider a more to-the-point statement: "FastJ's UI system".

File: src/main/java/tech/fastj/graphics/package-info.java [hunk 8]
```diff
@@ -0,0 +1,5 @@
+package tech.fastj.graphics;
+
+/**
+  Everything related to graphics and rendering in FastJ.
```
Comment (lucasstarsz) [inline src/main/java/tech/fastj/graphics/package-info.java:4]: Consider: "FastJ's Graphics and Rendering system".

File: src/main/java/tech/fastj/graphics/io/package-info.java [hunk 9]
```diff
@@ -0,0 +1,5 @@
+package tech.fastj.graphics.io;
+
+/**
+ * Code to read models and other data from files into FastJ.
```
Comment (lucasstarsz) [inline src/main/java/tech/fastj/graphics/io/package-info.java:4]: Consider: "FastJ model file IO".

File: src/main/java/tech/fastj/graphics/game/package-info.java [hunk 10]
```diff
@@ -0,0 +1,5 @@
+package tech.fastj.graphics.game;
+
+/**
+ * Different elements meant to be used in a FastJ game.
```
Comment (lucasstarsz) [inline src/main/java/tech/fastj/graphics/game/package-info.java:4]: This is a bit too broad of a topic -- "different elements" what elements? We can't tell anything about these elements other than that they should be used in a FastJ game, and _quite literally anything_ in FastJ's source could or maybe even should be used in a FastJ game.

Consider a more focused approach: "{\@code GameObject}s and specific instances".

File: src/main/java/tech/fastj/graphics/display/package-info.java [hunk 11]
```diff
@@ -0,0 +1,5 @@
+package tech.fastj.graphics.display;
+
+/**
+ * C
```
Comment (lucasstarsz) [inline src/main/java/tech/fastj/graphics/display/package-info.java:4]: This seems incomplete -- it should be removed alongside the other empty package infos.

File: src/main/java/tech/fastj/engine/package-info.java [hunk 12]
```diff
@@ -0,0 +1,5 @@
+package tech.fastj.engine;
+
+/**
+ * Classes used internally inside FastJ.
```
Comment (lucasstarsz) [inline src/main/java/tech/fastj/engine/package-info.java:4]: > internally inside

This is fairly redundant -- internal and inside both mean the same thing in this context.
Consider: "FastJ's internal engine classes".

**Review by lucasstarsz — CHANGES_REQUESTED** [review]
Not bad -- a few more changes to be made. I will say though, I would rather you remove the empty package info files. There's no point in having them if they're empty, which is what I was implying in the last review.

**Review by lucasstarsz — APPROVED** [review]
Looks good. Thanks for contributing to FastJ!

**Pull request merged**